backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/storage/disk/image/GetUnregisteredDisksQuery.java

```diff
@@ -78,7 +78,16 @@ protected void executeQueryCommand() {
             if (unregQueryReturn.getSucceeded()) {
                 unregisteredDisks.add(unregQueryReturn.getReturnValue());
             } else {
-                log.debug("Could not get populated disk: {}", unregQueryReturn.getExceptionString());
+                DiskImage returnValue = unregQueryReturn.getReturnValue();
+                if (returnValue.isDiskSnapshot() &&
+                        String.format("snapshot-%s", unregisteredDiskId).equals(returnValue.getDescription())) {
+                    log.debug("Could not get populated disk: {}", unregQueryReturn.getExceptionString());
+                    continue;
+                }
+
+                log.error("Could not get populated disk: {}", unregQueryReturn.getExceptionString());
+                getQueryReturnValue().setSucceeded(false);
+                break;
```

Comment:
not sure about the `break` here - it looks like rest-api doesn't check whether the query fails or not but displays whatever it returns, maybe for consistency we should continue here?

Reply:
if we continue don't we run the risk of having `succeeded` be set with `true` and have a partial disk list? (not sure what are the consequences of this though)

Reply:
in theory yes but in practice we usually don't call `setSucceeded` with `true` in queries and I don't see we do it here so that part seems fine